Test failures since version 0.13.2: `NameError: name 'dirty_equals' is not defined`

Our CI started failing after 0.13.2 release with

```
    def use_valid_old_values(old_value, new_value):
    
>       if isinstance(new_value, dirty_equals.DirtyEquals):
E       NameError: name 'dirty_equals' is not defined

.venv/lib64/python3.12/site-packages/inline_snapshot/_inline_snapshot.py:285: NameError
```

--- a/changelog.d/20240924_150856_15r10nk-git_fix_dirty_equals_bug.md
+++ b/changelog.d/20240924_150856_15r10nk-git_fix_dirty_equals_bug.md
@@ -1,0 +1,40 @@
+<!--
+A new scriv changelog fragment.
+
+Uncomment the section that is right (remove the HTML comment wrapper).
+-->
+
+<!--
+### Removed
+
+- A bullet item for the Removed category.
+
+-->
+<!--
+### Added
+
+- A bullet item for the Added category.
+
+-->
+<!--
+### Changed
+
+- A bullet item for the Changed category.
+
+-->
+<!--
+### Deprecated
+
+- A bullet item for the Deprecated category.
+
+-->
+### Fixed
+
+- removed non-optional dirty-equals dependency (#118)
+
+<!--
+### Security
+
+- A bullet item for the Security category.
+
+-->

--- a/src/inline_snapshot/_inline_snapshot.py
+++ b/src/inline_snapshot/_inline_snapshot.py
@@ -282,9 +282,6 @@ class EqValue(GenericValue):
 
         def use_valid_old_values(old_value, new_value):
 
-            if isinstance(new_value, dirty_equals.DirtyEquals):
-                assert False
-
             if (
                 isinstance(new_value, list)
                 and isinstance(old_value, list)

